docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,418 @@
+}
+
+func newWorkloadIdentityCredential(tenantID, clientID, file string, options *workloadIdentityCredentialOptions) (*workloadIdentityCredential, error) {
+	w := &workloadIdentityCredential{file: file}
+	cred, err := azidentity.NewClientAssertionCredential(tenantID, clientID, w.getAssertion, &azidentity.ClientAssertionCredentialOptions{ClientOptions: options.ClientOptions})
+	if err != nil {
+		return nil, err
+	}
+	w.cred = cred
+	return w, nil
+}
+
+func (w *workloadIdentityCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	return w.cred.GetToken(ctx, opts)
+}
+
+func (w *workloadIdentityCredential) getAssertion(context.Context) (string, error) {
+	if now := time.Now(); w.lastRead.Add(5 * time.Minute).Before(now) {
+		content, err := os.ReadFile(w.file)
+		if err != nil {
+			return "", err
+		}
+		w.assertion = string(content)
+		w.lastRead = now
+	}
+	return w.assertion, nil
+}
+
+```
+
+### <a name='OpenQuestions'></a>Open Questions
+
+#### <a name='Howtomultitenancy'></a>1. How to achieve multi-tenancy?
+
+The identity is tied to the client ID which can be supplied via AzureClusterIdentity. In case identity is not found in AzureClusterIdentity client ID from the service account annotation
+`azure.workload.identity/client-id` will be used.
+
+#### <a name='Howtodistributekeys'></a>2. How to distribute key pair to management cluster?
+
+Keys can be distributed by using the CABPK feature by creating a secret with name `<cluster-name>-sa` encompassing the key pair. More details on it [here](https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html)
+
+#### <a name='UserExperience'></a>3. User Experience
+
+Though AZWI has a lot of advantages as compared to AZWI, setting up AZWI involves couple of manual step and it can impact the user experience.
+
+### <a name='Limitations'></a>Limitations
+
+- Cloud provider azure does not support workload identity yet and it also needs to authenticate to azure APIs. 
+- That means one caveat of using AZWI in capz will be that using `UserAssignedIdentity` will become a requirement so that cloud provider azure is able to authenticate to the Azure APIs until cloud provider azure supports workload identity.
+
+### <a name='MigrationPlan'></a>Migration Plan
+
+Management clusters using AAD pod identity should have a seamless migration process which is well documented.
+
+For migrating an existing cluster to use AZWI following steps should be taken.
+
+- Generate a key pair or use the same key pairs as created via `<cluster-name>-sa` secret. 
+
+- Perform the following pre-requistes as discussed earlier in the **Bootstrap Cluster** section of the [document](#proposal):
+	- Generate and upload the JWKS and Discovery Document.
+	- Install the AZWI mutating admission webhook controller.
+	- Establish federated credential for the key pair. 	 
+- Create a `UserAssignedIdentity` and update the machine template to use user assigned identity.
```

Comment:
just making sure: this is mutable right?

Reply:
Oh, right! I think AzureMahcineTemplate is immutable. If it is, then we need to create a a new AzureMahcineTemplate and perform node roll outs. 
Let me validate this and update. Thanks